pandas/tests/tseries/holiday/test_federal.py

```diff
@@ -36,3 +37,15 @@ class MemorialDay(AbstractHolidayCalendar):
         datetime(1978, 5, 29, 0, 0),
         datetime(1979, 5, 28, 0, 0),
     ]
+
+
+def test_federal_holiday_inconsistent_returntype():
+    # GH 49075 test case
+    # Instantiate two calendars to rule out _cache
+    cal1 = USFederalHolidayCalendar()
+    cal2 = USFederalHolidayCalendar()
+
+    results_2018 = cal1.holidays(start=datetime(2018, 8, 1), end=datetime(2018, 8, 31))
+    results_2019 = cal2.holidays(start=datetime(2019, 8, 1), end=datetime(2019, 8, 31))
+
+    assert type(results_2018) == type(results_2019)
```

Comment:
Could you use `tm.assert_index_equal` or `tm.assert_series_equal` to compare the entire object?

Reply:
@mroeschke Thanks for bringing those to my attention, I can already think of ways to use those now.